File: NetBet/ApiControllers/BetController.fs
namespace NetBet.Controllers

open System
open System.Collections.Generic
open System.Linq
open Microsoft.AspNetCore.Mvc
open DbTypes
open DtoTypes

[<Route("api/[controller]/[action]")>]
[<ApiController>]
type BetController () =
    inherit ControllerBase()
    
    [<HttpGet("matchid")>]
    member __.GetBetsForMatch matchid =
        let bets = BetService.getBetsForMatch matchid
        ActionResult<BetWithOdds[]>(bets)

    [<HttpGet("eventid")>]
    member __.GetBetsForEvent eventid =
        let bets = BetService.getBetsForEvent eventid
        ActionResult<BetWithOdds[]>(bets)
        
    [<HttpGet("eventid")>]
    member __.GetPrettyBetsForEvent eventid = 
        let bets = BetService.getPrettyBetsForEvent eventid
        ActionResult<PrettyBet[]>(bets)

    [<HttpPost>]
    member __.PlaceSingleBet([<FromBody>] b: Bet) =
        BetService.placeSingleBet b

    [<HttpPost>]
    member __.PlaceParlayBet([<FromBody>] bets: Bet[]) = 
        BetService.placeParlayBet bets
        
    [<HttpPost>]
    member __.Delete([<FromBody>] b: Bet) =
        BetService.deleteBet b

    

File: NetBet.WebApp/ApiControllers/BetController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using static DbTypes;
using static DtoTypes;

namespace NetBet.WebApp.ApiControllers
{
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class BetController : ControllerBase
    {
        [HttpGet("{matchid}")]
        public BetWithOdds[] GetBetsForMatch(int matchid)
        {
            return BetService.getBetsForMatch(matchid);
        }
        
        [HttpGet("{eventid}")]
        public BetWithOdds[] GetBetsForEvent(int eventid)
        {
            return BetService.getBetsForEvent(eventid);
        }

        [HttpGet("{eventid}")]
        public PrettyBet[] GetPrettyBetsForEvent(int eventid)
        {
            return BetService.getPrettyBetsForEvent(eventid);
        }
        
        [HttpPost]
        public void PlaceBet([FromBody] PlaceBetDto dto)
        {
            BetService.placeBet(dto);
        }
        
        [HttpDelete()]
        public void Delete([FromBody] Bet b)
        {
            BetService.deleteBet(b);
        }

        [HttpGet("{seasonID}")]
        public SeasonWinPercent[] GetBetStatsForSeason(int seasonID)
        {
            return BetService.getBetStatsForSeason(seasonID);
        }
    }
}
